services/src/test/java/com/chatbot/services/HangoutsControllerTests.java

```diff
@@ -0,0 +1,166 @@
+    mockMvc = MockMvcBuilders.standaloneSetup(hangoutsController).build();
+    final ObjectMapper mapper = new ObjectMapper();
+    textMessageNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/textMessage.json"));
+    attachmentMessageNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/attachmentMessage.json"));
+    messageFromRoomNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/messageFromRoom.json"));
+    messageWithNoTypeNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/messageWithNoType.json"));
+    messageWithInvalidSenderNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/messageWithInvalidSender.json"));
+    messageWithNoSenderInfoNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/messageWithNoSenderInfo.json"));
+    messageWithCardClickNode = mapper
+        .readTree(HangoutsControllerTests.class
+        .getResourceAsStream("/cardClickedMessage.json"));
+  }
+
+  @Test
+  public void onRequest_textMessage() throws Exception {
+    final MvcResult mvcResult = mockMvc.perform(MockMvcRequestBuilders.post(uri)
+        .contentType(MediaType.APPLICATION_JSON_VALUE)
+        .header("user-agent", ChatServiceConstants.HANGOUTS_USER_AGENT)
+        .header("authorization", "Bearer 123")
+        .content(textMessageNode.toString())).andReturn();
+    assertEquals(200, mvcResult.getResponse().getStatus());
+    final ArgumentCaptor<ChatServiceRequest> request =
+        ArgumentCaptor.forClass(ChatServiceRequest.class);
+    verify(asyncService).chatServiceRequestHandler(request.capture());
+    assertEquals(ChatClient.HANGOUTS, request.getValue().getChatClient(),
+        "Error parsing chat client");
+    assertEquals("Hello", request.getValue().getUserMessage().getText(),
+        "Error parsing user message");
+    assertEquals("123456", request.getValue().getSender().getChatClientGeneratedId(),
+        "Error parsing chat client generated id");
+
+  }
+
+  @Test
+  public void onRequest_attachmentMessage() throws Exception {
+    mockMvc.perform(MockMvcRequestBuilders.post(uri)
+        .contentType(MediaType.APPLICATION_JSON_VALUE)
+        .header("user-agent", ChatServiceConstants.HANGOUTS_USER_AGENT)
+        .header("authorization", "Bearer 123")
+        .content(attachmentMessageNode.toString())).andReturn();
+    final ArgumentCaptor<ChatServiceRequest> request =
+        ArgumentCaptor.forClass(ChatServiceRequest.class);
+    verify(asyncService).chatServiceRequestHandler(request.capture());
+    assertEquals(1, request.getValue().getUserMessage().getAttachmentsCount(),
+        "Error parsing attachements");
+    assertEquals(MimeType.JPEG, request.getValue().getUserMessage().getAttachments(0).getMimeType(),
+        "Error parsing attachements");
```

Comment:
Error parsing mimetype?

Same for other instances

Reply:
updated